### DISCUSSION TITLE
Customize Sources: Filter "Section" -> Add options "has a value" / "has no value"

### DISCUSSION THREAD

**Discussion by ul8** [craftcms/cms#16270] [Ideas] [answered]
I need to configure a custom entry source to show only entries that have a section (i. e. all the things that used to be called entries in Craft 4). 

Right now I have to use the section filter and manually add ALL sections, which works but will inevitably fail once we add new sections and forget to add them to this filter.

It would be good to have "has a value" or "has no value" options here.

**Comment by brandonkelly** [comment, marked as answer]
Thanks for the suggestion!

Just made a couple improvements for Craft 5.6: (b8fe7e1f282a29be970b3b60d3e5d8f63e963406)

- “Section” condition rules now have a “has a value” operator option, which will filter out nested entries.
- “Matrix field” condition rules have been replaced with a new “Field” condition rule type, which includes all nested entry field types (Matrix + CKEditor, and possibly some third party field types) as options, and also has a “has a value” operator option for filtering out top-level entries.

**Reply by ul8** [reply]
Thanks so much!

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out with those changes 🎉
